kedro/framework/cli/jupyter.py

```diff
@@ -31,20 +31,56 @@
 OVERWRITE_HELP = """If Python file already exists for the equivalent notebook,
 overwrite its contents."""
 
+jupyter_command_group_order = ["init", "notebook", "lab", "convert"]
+jupyter_command_group_order = {
+    command: order for order, command in enumerate(jupyter_command_group_order)
+}
+
+
+class JupyterCommandGroup(click.Group):
+    """A custom class for ordering the `kedro jupyter` command groups"""
+
+    def __init__(self, name=None, commands=None, **attrs):
+        super().__init__(name, commands, **attrs)
+        #: the registered subcommands by their exported names.
+        self.commands = commands or {}
+
+    def list_commands(self, ctx):
+        """List commands according to a custom order"""
+        return sorted(self.commands, key=lambda k: jupyter_command_group_order[k])
+
 
 # pylint: disable=missing-function-docstring
 @click.group(name="Kedro")
 def jupyter_cli():  # pragma: no cover
     pass
 
 
-@jupyter_cli.group()
+@jupyter_cli.group(cls=JupyterCommandGroup)
 def jupyter():
     """Open Jupyter Notebook / Lab with project specific variables loaded, or
     convert notebooks into Kedro code.
     """
 
 
+@forward_command(jupyter, "init", forward_help=True)
+@env_option
+@click.pass_obj  # this will pass the metadata as first argument
+def init(
+    metadata: ProjectMetadata,
+    env,
+    args,
+    **kwargs,
+):  # pylint: disable=unused-argument
+    """Initialise the Jupyter Kernel for a kedro project."""
+    _check_module_importable("notebook")
```

Comment:
It probably won't make any difference in practice, since anyone executing this command will probably have it installed... but I don't think `notebook` is actually necessary for making a Jupyter kernel. Looking at `_create_kernel` we should probably be checking `ipykernel` instead.